src/support_enumeration.jl

```diff
@@ -0,0 +1,252 @@
+#=
+Compute all mixed Nash equilibria of a 2-player (non-degenerate) normal
+form game by support enumeration.
+
+References
+----------
+B. von Stengel, "Equilibrium Computation for Two-Player Games in
+Strategic and Extensive Form," Chapter 3, N. Nisan, T. Roughgarden, E.
+Tardos, and V. Vazirani eds., Algorithmic Game Theory, 2007.
+=#
+
+"""
+    support_enumeration(g::NormalFormGame)
+
+Compute mixed-action Nash equilibria with equal support size for a
+2-player normal form game by support enumeration. For a
+non-degenerate game input, these are all Nash equilibria.
+
+The algorithm checks all the equal-size support pairs; if the
+players have the same number n of actions, there are 2n choose n
+minus 1 such pairs. This should thus be used only for small games.
+
+# Arguments
+* `g::NormalFormGame`: NormalFormGame instance.
+
+# Returns
+* `::Vector{Tuple{Vector{Float64},Vector{Float64}}}`: Mixed-action
+    Nash equilibria that are found.
+"""
+function support_enumeration(g::NormalFormGame)
+
+    task = support_enumeration_task(g)
+
+    NEs = Tuple{Vector{Float64},Vector{Float64}}[NE for NE in task]
+
+    return NEs
+
+end
+
+"""
+    support_enumeration_task(g::NormalFormGame)
+
+Task version of `support_enumeration`.
+
+# Arguments
+* `g::NormalFormGame`: NormalFormGame instance.
+
+# Returns
+* `::Task`: runnable task for generating Nash equilibria.
+"""
+function support_enumeration_task(g::NormalFormGame)
+
+    N = length(g.nums_actions)
+    if N != 2
+        throw(ArgumentError("Implemented only for 2-player games"))
+    end
+
+    task = Task(() -> _support_enumeration_task(g.players[1].payoff_array,
+                                                g.players[2].payoff_array))
+
+    return task
+end
+
+"""
+    _support_enumeration_task{T<:Real}(payoff_matrix1::Matrix{T},
+                                       payoff_matrix2::Matrix{T})
+
+Main body of `support_enumeration_task`.
+
+# Arguments
+* `payoff_matrix1::Matrix{T}`: Payoff matrix of player 1.
+* `payoff_matrix2::Matrix{T}`: Payoff matrix of player 2.
+
+# Produces
+* `Tuple{Vector{Float64},Vector{Float64}}`: Tuple of Nash equilibrium
```

Comment:
Contrary to what is written in the [manual](http://docs.julialang.org/en/stable/stdlib/linalg/#Base.A_ldiv_B!), the following doesn't work:

```jl
julia> A = [1 2; 7 5//1]
2×2 Array{Rational{Int64},2}:
 1//1  2//1
 7//1  5//1

julia> b = [1, 1//1]
2-element Array{Rational{Int64},1}:
 1//1
 1//1

julia> A_ldiv_B!(factorize(A), b)
2-element Array{Rational{Int64},1}:
 -1//3
  2//3

julia> b
2-element Array{Rational{Int64},1}:
 1//1
 1//1
```

where the solution should have been contained in `b`. The same problem occurs with `A_ldiv_B!(sol, factorize(A), b)` with preallocated `sol`.

Do you see what is happening? @sglyon @cc7768